rest-api/alembic/versions/45d77ae13992_add_id_and_sample_history_tables.py

```diff
@@ -0,0 +1,78 @@
+"""add id and sample history tables
+
+Revision ID: 45d77ae13992
+Revises: 4aa79d5ac86e
+Create Date: 2018-08-31 10:18:06.252823
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import model.utils
+
+
+from participant_enums import PhysicalMeasurementsStatus, QuestionnaireStatus, OrderStatus
+from participant_enums import WithdrawalStatus, SuspensionStatus, QuestionnaireDefinitionStatus
+from participant_enums import EnrollmentStatus, Race, SampleStatus, OrganizationType, BiobankOrderStatus
+from participant_enums import MetricSetType, MetricsKey
+from model.site_enums import SiteStatus, EnrollingStatus, DigitalSchedulingStatus, ObsoleteStatus
+from model.code import CodeType
+
+# revision identifiers, used by Alembic.
+revision = '45d77ae13992'
+down_revision = '4aa79d5ac86e'
+branch_labels = None
+depends_on = None
+
+
+def upgrade(engine_name):
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('biobank_order_identifier_history',
+    sa.Column('version', sa.Integer(), nullable=False),
+    sa.Column('system', sa.String(length=80), nullable=False),
+    sa.Column('value', sa.String(length=80), nullable=False),
+    sa.Column('biobank_order_id', sa.String(length=80), nullable=False),
+    sa.ForeignKeyConstraint(['biobank_order_id'], ['biobank_order.biobank_order_id'], ),
+    sa.PrimaryKeyConstraint('version', 'system', 'value')
```

Comment:
The primary key is in the wrong order here; should be system, value, version. (Which kinda matters for querying on this.) I believe if you get rid of the @declared_attr stuff below it should work. You *may* need to add a version column to BiobankOrderIdentifierBase or BiobankOrderedSampleBase, or override all the primary key columns in the subclasses to get SQLAlchemy to handle this appropriately... I'm not sure.

Alternatively, you can mess with the generated code here directly... just make sure that SQLAlchemy doesn't do something crazy like try to change the primary key if you run generate_schema again afterwards.

Reply:
done